### ISSUE TITLE
Error raised with empty data resource collection

### ISSUE THREAD

**Issue by jkburges** [trainline/json-api-vanilla#2]
e.g. the following:

```ruby
pry> JSON::Api::Vanilla.parse('{ "data": [] }')                                                             
JSON::Api::Vanilla::InvalidRootStructure: JSON:API document must contain at least one of these objects: data, errors, meta
```

even though I believe the payload *is* valid JSON API:

<blockquote>
A document MUST contain at least one of the following top-level members:

* data: the document’s “primary data”
* errors: an array of error objects
* meta: a meta object that contains non-standard meta-information.

The members data and errors MUST NOT coexist in the same document.

A document MAY contain any of these top-level members:

jsonapi: an object describing the server’s implementation
links: a links object related to the primary data.
included: an array of resource objects that are related to the primary data and/or each other (“included resources”).
If a document does not contain a top-level data key, the included member MUST NOT be present either.

*snip*

Primary data MUST be either:

* a single resource object, a single resource identifier object, or null, for requests that target single resources
* an array of resource objects, an array of resource identifier objects, or **an empty array ([])** (*my emphasis*), for requests that target resource collections
</blockquote>

Ref: http://jsonapi.org/format/

**Comment by espadrine** [comment]
Thank you for reporting this bug! It should be fixed now, in version 1.0.1 (published a few seconds ago).

**Comment by jkburges** [comment]
I can confirm it works, you're a legend.  Nice gem btw.
